build-tools/packages/build-tools/src/common/biome2Config.ts

```diff
@@ -0,0 +1,396 @@
+ * For each parent config found, recursively resolves any `extends` declarations.
+ *
+ * @param startDir - The directory containing the child config file. Parent discovery starts
+ *                   from this directory's parent (i.e., startDir itself is not searched).
+ * @returns Array of config paths in order from root to nearest parent (not including the starting directory),
+ *          with all extends chains resolved
+ */
+async function findParentBiome2Configs(startDir: string): Promise<string[]> {
+	// Collect configs in child-to-root order, then reverse at the end
+	const configsChildToRoot: string[][] = [];
+	// Start from parent directory - we don't want to include the config in startDir
+	let currentDir = path.dirname(startDir);
+	const fsRoot = path.parse(startDir).root;
+
+	while (currentDir !== fsRoot) {
+		const configPath = await findBiome2ConfigInDirectory(currentDir);
+		if (configPath) {
+			const config = await loadRawBiome2Config(configPath);
+
+			// Recursively resolve any extends for this parent config using shared utility
+			const parentConfigPaths = await resolveExtendsChainGeneric(
+				configPath,
+				loadRawBiome2Config,
+			);
+
+			// Add to the list (we'll reverse at the end)
+			configsChildToRoot.push(parentConfigPaths);
+
+			// If this config has root: true, stop walking up
+			if (config.root === true) {
+				break;
+			}
+		}
+		currentDir = path.dirname(currentDir);
+	}
+
+	// Reverse and flatten to get root-to-child order
+	return configsChildToRoot.reverse().flat();
+}
+
+/**
+ * Looks for a Biome config file in the given directory.
+ * @returns The path to the config file if found, undefined otherwise.
+ */
+async function findBiome2ConfigInDirectory(dir: string): Promise<string | undefined> {
+	const possibleNames = ["biome.json", "biome.jsonc"];
+	for (const name of possibleNames) {
+		const configPath = path.join(dir, name);
+		try {
+			await stat(configPath);
+			return configPath;
+		} catch {
+			// File doesn't exist, try next
+		}
+	}
+	return undefined;
+}
+
+/**
+ * Loads a Biome 2.x configuration file. If the config extends others or has parent configs in the directory tree,
+ * those are loaded recursively and the results are merged. Array-type values are not merged, in accordance with
+ * how Biome applies configs.
+ *
+ * @remarks
+ *
+ * The intent is to merge the configs in the same way that Biome itself does, but the implementation is based on the
+ * Biome documentation, so there may be subtle differences unaccounted for. Where this implementation diverges from
+ * Biome's behavior, this function should be considered incorrect.
+ *
+ * Relevant Biome documentation: {@link https://biomejs.dev/guides/big-projects/}
+ */
+export async function loadBiome2Config(configPath: string): Promise<Biome2ConfigResolved> {
+	const allConfigPaths = await getAllBiome2ConfigPaths(configPath);
+	return loadBiome2Configs(allConfigPaths);
+}
```

Comment:
It's confusing why we have two functions suimiarly named and the only difference is that one takes a string and the other a string array. Should they have different names to make it clear what their purpose is? tsdoc param docs would help.